### PR TITLE
Resolves #2056

### LINKED ISSUE #2056: Copy Button also picks up command prompt
[Enter feedback here]

When looking at the commands, the copy buttom copies everything : 

```
C:\PS>Get-ADUser -Filter * -SearchBase "OU=Finance,OU=UserAccounts,DC=FABRIKAM,DC=COM"
```

It should only be copying the command.

```
Get-ADUser -Filter * -SearchBase "OU=Finance,OU=UserAccounts,DC=FABRIKAM,DC=COM"
```

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2a4eb77a-834c-0412-8f9f-9c515172237e
* Version Independent ID: 0228a754-2177-71d0-92cc-ae144070947e
* Content: [Get-ADUser (activedirectory)](https://docs.microsoft.com/en-us/powershell/module/activedirectory/get-aduser?view=winserver2012-ps)
* Content Source: [docset/winserver2012-ps/activedirectory/Get-ADUser.md](https://github.com/MicrosoftDocs/windows-powershell-docs/blob/master/docset/winserver2012-ps/activedirectory/Get-ADUser.md)
* Product: **w10**
* Technology: **powershell**
* GitHub Login: @andreabarr
* Microsoft Alias: **v-anbarr**

### PR DESCRIPTION
**Pull request by velkovb**
Fix examples.
https://github.com/MicrosoftDocs/windows-powershell-docs/issues/2056

### CODE REVIEW

**Comment by e0i** [comment]
@dariomws @get-itips @tseward

Peer review needed for this PR.

cc @JohanFreelancer9 for copy editing.

Thank you.

**Review by dariomws — APPROVED** [review]

**Review by get-itips — COMMENTED** [review]
I am ok with it, only one thing, are we going to leave cmdlet's output on examples? Is this vital for the article? @tseward @dariomws

**Comment by dariomws** [comment]
@get-itips not sure about what we usually do here, but for me looks better without them because you can copy and paste the command succesfully.

**Comment by velkovb** [comment]
@dariomws @get-itips  Some will complain that there is no example of the output of the get command, some that when there is output they cannot copy and paste :) I think it is useful to see the output but somehow be able to copy only the command, though I do not think it is currently possible.

**Comment by dariomws** [comment]
> @dariomws @get-itips Some will complain that there is no example of the output of the get command, some that when there is output they cannot copy and paste :) I think it is useful to see the output but somehow be able to copy only the command, though I do not think it is currently possible.

Maybe we can put the output in the description of the example. @get-itips @velkovb What do you think?

**Comment by get-itips** [comment]
> Maybe we can put the output in the description of the example. @get-itips @velkovb What do you think?
Has that been done in the past?

File: docset/winserver2012-ps/activedirectory/Get-ADUser.md [hunk 1]
```diff
@@ -37,104 +37,79 @@ Get-ADUser [-AuthType <ADAuthType>] [-Credential <PSCredential>] -LDAPFilter <St
...
 
 ## DESCRIPTION
-The Get-ADUser cmdlet gets a user object or performs a search to retrieve multiple user objects.
+The **Get-ADUser** cmdlet gets a specified user object or performs a search to get multiple user objects.
 
-The Identity parameter specifies the Active Directory user to get.
-You can identify a user by its distinguished name (DN), GUID, security identifier (SID), Security Accounts Manager (SAM) account name or name.
-You can also set the parameter to a user object variable, such as $\<localUserObject\> or pass a user object through the pipeline to the Identity parameter.
+The *Identity* parameter specifies the Active Directory user to get.
+You can identify a user by its distinguished name (DN), GUID, security identifier (SID), Security Account Manager (SAM) account name or name.
```
Comment (JohanFreelancer9) [inline docset/winserver2012-ps/activedirectory/Get-ADUser.md:43]:
```suggestion
You can identify a user by its distinguished name (DN), GUID, security identifier (SID), Security Account Manager (SAM) account name, or name.
```

File: docset/winserver2012-ps/activedirectory/Get-ADUser.md [hunk 2]
```diff
@@ -37,104 +37,79 @@ Get-ADUser [-AuthType <ADAuthType>] [-Credential <PSCredential>] -LDAPFilter <St
...
+You can also set the parameter to a user object variable, such as `$<localUserObject>` or pass a user object through the pipeline to the *Identity* parameter.
```
Comment (JohanFreelancer9) [inline docset/winserver2012-ps/activedirectory/Get-ADUser.md:44]:
```suggestion
You can also set the parameter to a user object variable such as `$<localUserObject>` or pass a user object through the pipeline to the *Identity* parameter.
```

File: docset/winserver2012-ps/activedirectory/Get-ADUser.md [hunk 3]
```diff
@@ -37,104 +37,79 @@ Get-ADUser [-AuthType <ADAuthType>] [-Credential <PSCredential>] -LDAPFilter <St
...
+The *Filter* parameter uses the PowerShell Expression Language to write query strings for Active Directory.
+PowerShell Expression Language syntax provides rich type conversion support for value types received by the *Filter* parameter.
+For more information about the *Filter* parameter syntax, type `Get-Help about_ActiveDirectory_Filter`.
+If you have existing Lightweight Directory Access Protocol (LDAP) query strings, you can use the *LDAPFilter* parameter.
 
 This cmdlet retrieves a default set of user object properties.
-To retrieve additional properties use the Properties parameter.
-For more information about the how to determine the properties for user objects, see the Properties parameter description.
+To retrieve additional properties use the **Properties** parameter.
+For more information about the how to determine the properties for user objects, see the **Properties** parameter description.
```
Comment (JohanFreelancer9) [inline docset/winserver2012-ps/activedirectory/Get-ADUser.md:54]:
```suggestion
For more information about how to determine the properties for user objects, see the **Properties** parameter description.
```

File: docset/winserver2012-ps/activedirectory/Get-ADUser.md [hunk 4]
```diff
@@ -37,104 +37,79 @@ Get-ADUser [-AuthType <ADAuthType>] [-Credential <PSCredential>] -LDAPFilter <St
...
+```powershell
+PS C:\> Get-ADUser -Filter "Name -eq 'ChewDavid'" -SearchBase "DC=AppNC" -Properties "mail" -Server lds.Fabrikam.com:50000
 ```
 
-Description
-
------------
-
-Get the user with name 'GlenJohn' on the AD LDS instance.
+This command gets the user with name ChewDavid in the Active Directory Lightweight Directory Services (AD LDS) instance.
```
Comment (JohanFreelancer9) [inline docset/winserver2012-ps/activedirectory/Get-ADUser.md:99]:
```suggestion
This command gets the user with the name ChewDavid in the Active Directory Lightweight Directory Services (AD LDS) instance.
```

File: docset/winserver2012-ps/activedirectory/Get-ADUser.md [hunk 5]
```diff
@@ -37,104 +37,79 @@ Get-ADUser [-AuthType <ADAuthType>] [-Credential <PSCredential>] -LDAPFilter <St
...
+- Basic or 1
 
 The default authentication method is Negotiate.
 
-A Secure Sockets Layer (SSL) connection is required for the Basic authentication method.
-
-The following example shows how to set this parameter to Basic.
-
--AuthType Basic
+A Secure Sockets Layer (SSL) connection is required for the Basic authentication method.s
```
Comment (JohanFreelancer9) [inline docset/winserver2012-ps/activedirectory/Get-ADUser.md:112]:
```suggestion
A Secure Sockets Layer (SSL) connection is required for the Basic authentication method.
```

File: docset/winserver2012-ps/activedirectory/Get-ADUser.md [hunk 6]
```diff
@@ -154,19 +129,13 @@ Specifies the user account credentials to use to perform this task.
...
-$AdminCredentials = Get-Credential "Domain01\User01"
-
-The following shows how to set the Credential parameter to these credentials.
+You can also create a **PSCredential** object by using a script or by using the **Get-Credential** cmdlet.
+You can then set the *Credential* parameter to the **PSCredential** object.
 
--Credential $AdminCredentials
-
-If the acting credentials do not have directory-level permission to perform the task, Active Directory PowerShell returns a terminating error.
+If the acting credentials do not have directory-level permission to perform the task, Active Directory PowerShell returns a terminating error.s
```
Comment (JohanFreelancer9) [inline docset/winserver2012-ps/activedirectory/Get-ADUser.md:138]:
```suggestion
If the acting credentials do not have directory-level permission to perform the task, Active Directory PowerShell returns a terminating error.
```

File: docset/winserver2012-ps/activedirectory/Get-ADUser.md [hunk 7]
```diff
@@ -276,35 +200,17 @@ Accept wildcard characters: False
...
-This parameter can also get this object through the pipeline or you can set this parameter to an object instance.
-
-This example shows how to set the parameter to a distinguished name.
-
--Identity  "CN=SaraDavis,CN=Europe,CN=Users,DC=corp,DC=contoso,DC=com"
-
-This example shows how to set this parameter to a user object instance named "userInstance".
-
--Identity   $userInstance
+This parameter can also get this object through the pipeline or you can set this parameter to an object instance.s
```
Comment (JohanFreelancer9) [inline docset/winserver2012-ps/activedirectory/Get-ADUser.md:213]:
```suggestion
This parameter can also get this object through the pipeline or you can set this parameter to an object instance.
```

File: docset/winserver2012r2-ps/addsadministration/Get-ADUser.md [hunk 8]
```diff
@@ -47,15 +47,15 @@ Get-ADUser [-AuthType <ADAuthType>] [-Credential <PSCredential>] -LDAPFilter <St
 ## DESCRIPTION
 The **Get-ADUser** cmdlet gets a specified user object or performs a search to get multiple user objects.
 
-The **Identity** parameter specifies the Active Directory user to get.
+The *Identity* parameter specifies the Active Directory user to get.
 You can identify a user by its distinguished name (DN), GUID, security identifier (SID), Security Account Manager (SAM) account name or name.
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/addsadministration/Get-ADUser.md:51]:
```suggestion
You can identify a user by its distinguished name (DN), GUID, security identifier (SID), Security Account Manager (SAM) account name, or name.
```

File: docset/winserver2012r2-ps/addsadministration/Get-ADUser.md [hunk 9]
```diff
@@ -100,8 +100,8 @@ DistinguishedName : CN=Chew David,OU=NorthAmerica,OU=Sales,OU=UserAccounts,DC=FA
 This command gets all of the properties of the user with the SAM account name ChewDavid.
 
 ### Example 4: Get a specified user
-```
-PS C:\>Get-ADUser -Filter "Name -eq 'ChewDavid'" -SearchBase "DC=AppNC" -Properties "mail" -Server lds.Fabrikam.com:50000
+```powershell
+PS C:\> Get-ADUser -Filter "Name -eq 'ChewDavid'" -SearchBase "DC=AppNC" -Properties "mail" -Server lds.Fabrikam.com:50000
 ```
 
 This command gets the user with name ChewDavid in the Active Directory Lightweight Directory Services (AD LDS) instance.
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/addsadministration/Get-ADUser.md:107]:
```suggestion
This command gets the user with the name ChewDavid in the Active Directory Lightweight Directory Services (AD LDS) instance.
```

File: docset/winserver2012r2-ps/addsadministration/Get-ADUser.md [hunk 10]
```diff
@@ -47,15 +47,15 @@ Get-ADUser [-AuthType <ADAuthType>] [-Credential <PSCredential>] -LDAPFilter <St
...
+You can also set the parameter to a user object variable, such as `$<localUserObject>` or pass a user object through the pipeline to the *Identity* parameter.
 
-To search for and retrieve more than one user, use the **Filter** or **LDAPFilter** parameters.
-The **Filter** parameter uses the PowerShell Expression Language to write query strings for Active Directory.
-PowerShell Expression Language syntax provides rich type conversion support for value types received by the **Filter** parameter.
-For more information about the **Filter** parameter syntax, type `Get-Help about_ActiveDirectory_Filter`.
-If you have existing Lightweight Directory Access Protocol (LDAP) query strings, you can use the **LDAPFilter** parameter.
+To search for and retrieve more than one user, use the *Filter* or *LDAPFilter* parameters.
+The *Filter* parameter uses the PowerShell Expression Language to write query strings for Active Directory.
+PowerShell Expression Language syntax provides rich type conversion support for value types received by the *Filter* parameter.
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/addsadministration/Get-ADUser.md:56]:
```suggestion
PowerShell Expression Language syntax provides rich type-conversion support for value types received by the *Filter* parameter.
```

File: docset/winserver2012r2-ps/addsadministration/Get-ADUser.md [hunk 11]
```diff
@@ -250,25 +253,25 @@ Accept wildcard characters: False
...
+The cmdlet searches this partition to find the object defined by the *Identity* parameter.
 
-In many cases, a default value is used for the **Partition** parameter if no value is specified.
+In many cases, a default value is used for the *Partition* parameter if no value is specified.
 The rules for determining the default value are given below.
-Note that rules listed first are evaluated first and once a default value can be determined, no further rules are evaluated.
+Note that rules listed first are evaluated first and when a default value can be determined, no further rules are evaluated.
 
-In Active Directory Domain Services (AD DS) environments, a default value for **Partition** is set in the following cases: 
+In AD DS environments, a default value for *Partition* are set in the following cases:
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/addsadministration/Get-ADUser.md:262]:
```suggestion
In AD DS environments, a default value for *Partition* is set in the following cases: 
```

File: docset/winserver2012r2-ps/addsadministration/Get-ADUser.md [hunk 12]
```diff
@@ -393,27 +396,25 @@ Accept wildcard characters: False
 ```
 
 ### -Server
-Specifies the AD DS instance to connect to, by providing one of the following values for a corresponding domain name or directory server.
-The service may be any of the following: AD LDS, AD DS, or Active Directory snapshot instance.
-
-Specify the AD DS instance in one of the following ways:  
+Specifies the Active Directory Domain Services instance to connect to, by providing one of the following values for a corresponding domain name or directory server.
+The service may be any of the following: Active Directory Lightweight Domain Services, Active Directory Domain Services or Active Directory Snapshot instance.
```
Comment (JohanFreelancer9) [inline docset/winserver2012r2-ps/addsadministration/Get-ADUser.md:400]:
```suggestion
The service may be any of the following: Active Directory Lightweight Domain Services, Active Directory Domain Services, or Active Directory Snapshot instance.
```

File: docset/winserver2012-ps/activedirectory/Get-ADUser.md [hunk 13]
```diff
@@ -37,104 +37,79 @@ Get-ADUser [-AuthType <ADAuthType>] [-Credential <PSCredential>] -LDAPFilter <St
...
+You can also set the parameter to a user object variable, such as `$<localUserObject>` or pass a user object through the pipeline to the *Identity* parameter.
 
-To search for and retrieve more than one user, use the Filter or LDAPFilter parameters.
-The Filter parameter uses the PowerShell Expression Language to write query strings for Active Directory.
-PowerShell Expression Language syntax provides rich type conversion support for value types received by the Filter parameter.
-For more information about the Filter parameter syntax, see about_ActiveDirectory_Filter.
-If you have existing LDAP query strings, you can use the LDAPFilter parameter.
+To search for and retrieve more than one user, use the *Filter* or *LDAPFilter* parameters.
+The *Filter* parameter uses the PowerShell Expression Language to write query strings for Active Directory.
+PowerShell Expression Language syntax provides rich type conversion support for value types received by the *Filter* parameter.
```
Comment (JohanFreelancer9) [inline docset/winserver2012-ps/activedirectory/Get-ADUser.md:48]:
```suggestion
PowerShell Expression Language syntax provides rich type-conversion support for value types received by the *Filter* parameter.
```

**Review by JohanFreelancer9 — CHANGES_REQUESTED** [review]
Copy review.

**Review by JohanFreelancer9 — APPROVED** [review]

**Review by JohanFreelancer9 — APPROVED** [review]

**Comment by dariomws** [comment]
> > Maybe we can put the output in the description of the example. @get-itips @velkovb What do you think?
> 
> Has that been done in the past?

I don't remember It's true, but I saw that some cmdlets have a explanation or example of the output in the cmdlet description section.

**Comment by get-itips** [comment]
Let's leave it as is..

**Review by get-itips — APPROVED** [review]

**Review by tseward — APPROVED** [review]

**Review by JohanFreelancer9 — APPROVED** [review]

**Comment by velkovb** [comment]
@get-itips @dariomws So here is what I recently found on the output question:
[PowerShell-Docs style guide](https://docs.microsoft.com/PowerShell/scripting/community/contributing/powershell-style-guide#executable-examples)

```powershell
Get-Command -Module Microsoft.PowerShell.Security
```

```Output
CommandType  Name                        Version    Source
-----------  ----                        -------    ------
Cmdlet       ConvertFrom-SecureString    3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       ConvertTo-SecureString      3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Get-Acl                     3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Get-AuthenticodeSignature   3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Get-CmsMessage              3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Get-Credential              3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Get-ExecutionPolicy         3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Get-PfxCertificate          3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       New-FileCatalog             3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Protect-CmsMessage          3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Set-Acl                     3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Set-AuthenticodeSignature   3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Set-ExecutionPolicy         3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Test-FileCatalog            3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Unprotect-CmsMessage        3.0.0.0    Microsoft.PowerShell.Security
```

I will start doing this in PRs :)

**Comment by dariomws** [comment]
Testing:

```powershell
Get-Command -Module Microsoft.PowerShell.Security
```

```Output
CommandType  Name                        Version    Source
-----------  ----                        -------    ------
Cmdlet       ConvertFrom-SecureString    3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       ConvertTo-SecureString      3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Get-Acl                     3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Get-AuthenticodeSignature   3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Get-CmsMessage              3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Get-Credential              3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Get-ExecutionPolicy         3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Get-PfxCertificate          3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       New-FileCatalog             3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Protect-CmsMessage          3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Set-Acl                     3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Set-AuthenticodeSignature   3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Set-ExecutionPolicy         3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Test-FileCatalog            3.0.0.0    Microsoft.PowerShell.Security
Cmdlet       Unprotect-CmsMessage        3.0.0.0    Microsoft.PowerShell.Security
```

**Comment by dariomws** [comment]
@velkovb like it! Let's try, I think we should be able to copy the command separately in this way.

Thanks!

**Comment by velkovb** [comment]
@dariomws I created a new PR to implement that - https://github.com/MicrosoftDocs/windows-powershell-docs/pull/2190
@get-itips @e0i

**Pull request merged**